src/commands/edge_app_manifest.rs

```diff
@@ -0,0 +1,549 @@
+use crate::commands::CommandError;
+use std::collections::HashMap;
+
+use std::fs;
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+use crate::commands::edge_app_settings::{Setting, serialize_settings, deserialize_settings};
+use crate::commands::serde_utils::{deserialize_option_string_field, string_field_is_none_or_empty};
+
+#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct EdgeAppManifest {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub app_id: Option<String>,
+    #[serde(
+        deserialize_with = "deserialize_user_version",
+        skip_serializing_if = "string_field_is_none_or_empty",
+        default
+    )]
+    pub user_version: Option<String>,
+    #[serde(
+        deserialize_with = "deserialize_description",
+        skip_serializing_if = "string_field_is_none_or_empty",
+        default
+    )]
+    pub description: Option<String>,
+    #[serde(
+        deserialize_with = "deserialize_icon",
+        skip_serializing_if = "string_field_is_none_or_empty",
+        default
+    )]
+    pub icon: Option<String>,
+    #[serde(
+        deserialize_with = "deserialize_author",
+        skip_serializing_if = "string_field_is_none_or_empty",
+        default
+    )]
+    pub author: Option<String>,
+    #[serde(
+        deserialize_with = "deserialize_homepage_url",
+        skip_serializing_if = "string_field_is_none_or_empty",
+        default
+    )]
+    pub homepage_url: Option<String>,
+    #[serde(
+        serialize_with = "serialize_settings",
+        deserialize_with = "deserialize_settings",
+        default
+    )]
+    pub settings: Vec<Setting>,
+}
+
+fn deserialize_user_version<'de, D>(deserializer: D) -> Result<Option<String>, D::Error>
+where
+    D: serde::de::Deserializer<'de>,
+{
```

Comment:
we don't put the brace on the next line anywhere else... can you fix it? and check that formatter is ok with that

Reply:
I copied formatting from the serialize/deserialize functions that were in the code before, I'll fix it.

Which formatter do you mean?

Reply:
cargo fmt, I'll take a note of that too